src/Microsoft.InsightsGenerator/RulesEngine.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using System.Diagnostics;
+using System.IO;
+
+namespace Microsoft.InsightsGenerator
+{ 
+    public class RulesEngine
+    {
+        public class ColoumnHeaders
```

Comment:
Standard .Net coding conventions discourage having public fields, and recommends using properties for this purpose.  Also, public properties should use a capital letter for first char following PascalCase naming convention.  For example:

```C#
public List<string> SingleHashValues { get; set; }
// etc..
```